lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3136,14 +3136,13 @@ def errorbar(self, x, y, z, zerr=None, yerr=None, xerr=None, fmt='',
         if ecolor is None:
             ecolor = base_style['color']
 
-        # Eject any marker information from line format string, as it's not
+        # Eject any line-specific information from format string, as it's not
         # needed for bars or caps.
-        base_style.pop('marker', None)
-        base_style.pop('markersize', None)
-        base_style.pop('markerfacecolor', None)
-        base_style.pop('markeredgewidth', None)
-        base_style.pop('markeredgecolor', None)
-        base_style.pop('linestyle', None)
+        for key in ['marker', 'markersize', 'markerfacecolor',
+                    'markeredgewidth', 'markeredgecolor', 'markevery',
+                    'linestyle', 'fillstyle', 'drawstyle', 'dash_capstyle',
+                    'dash_joinstyle', 'solid_capstyle', 'solid_joinstyle']:
```

Comment:
If these two lists are the same, is it worth defining them as a variable somewhere so they can be shared?

Reply:
Well, @anntzer had said he wanted to do some refactoring of `errorbar`; hoping that would take care of the duplication of the entire function.